sigmastate/src/main/scala/sigmastate/lang/Terms.scala

```diff
@@ -140,29 +140,20 @@ object Terms {
 
     protected final override def eval(env: DataEnv)(implicit E: ErgoTreeEvaluator): Any = {
       addCost(Apply.costKind)
-      if (args.isEmpty) {
-        // TODO coverage
-        val fV = func.evalTo[() => Any](env)
-        fV()
-      }
-      else if (args.length == 1) {
+      if (args.length == 1) {
```

Comment:
ScalaDoc missed for both `eval` method and `Apply` class. Also, add comment why new behavior is non-breaking. And comment args limits  etc